src/components/render-page/window-post-message-communicator/window-post-message-communicator.ts

```diff
@@ -9,19 +9,39 @@
  */
 export class IframeCommunicatorSendingError extends Error {}
 
+export type Handler<MESSAGES, MESSAGE_TYPE extends string> =
+  | ((values: Extract<MESSAGES, PostMessage<MESSAGE_TYPE>>) => void)
+  | undefined
+
+export type HandlerMap<MESSAGES, MESSAGE_TYPE extends string> = Partial<{
+  [key in MESSAGE_TYPE]: Handler<MESSAGES, MESSAGE_TYPE>
+}>
+
+export interface PostMessage<MESSAGE_TYPE extends string> {
+  type: MESSAGE_TYPE
+}
+
 /**
  * Base class for communication between renderer and editor.
```

Comment:
Shouldn't this add the parameters here?

Reply:
Why should it? The most basic message you have has no parameters. If you take a look into render-messages.ts then you'll find the interfaces for the specific messages with parameters that extend this interface.

Reply:
Fair. I missed the `abstract`